src/durableorchestrationclient.ts

```diff
@@ -362,39 +368,51 @@ export class DurableOrchestrationClient {
-        let requestUrl = this.clientData.managementUrls.sendEventPostUri
-            .replace(idPlaceholder, instanceId)
-            .replace(this.eventNamePlaceholder, eventName);
+        let requestUrl: string;
+        if (this.clientData.rpcBaseUrl) {
+            // Fast local RPC path
+            let path = `instances/${instanceId}/raiseEvent/${eventName}`;
+            const query: string[] = [];
+            if (taskHubName) { query.push(`taskHub=${taskHubName}`); }
+            if (connectionName) { query.push(`connection=${connectionName}`); }
 
-        if (taskHubName) {
-            requestUrl = requestUrl.replace(this.clientData.taskHubName, taskHubName);
-        }
+            if (query.length > 0) {
+                path += "?" + query.join("&");
+            }
 
-        if (connectionName) {
-            requestUrl = requestUrl.replace(/(connection=)([\w]+)/gi, "$1" + connectionName);
-        }
+            requestUrl = new URL(path, this.clientData.rpcBaseUrl).href;
+        } else {
+            // Legacy app frontend path
+            const idPlaceholder = this.clientData.managementUrls.id;
+            requestUrl = this.clientData.managementUrls.sendEventPostUri
+                .replace(idPlaceholder, instanceId)
+                .replace(this.eventNamePlaceholder, eventName);
+
+            if (taskHubName) {
+                requestUrl = requestUrl.replace(this.clientData.taskHubName, taskHubName);
+            }
 
-        try {
-            const response = await this.axiosInstance.post(requestUrl, JSON.stringify(eventData));
-            switch (response.status) {
-                case 202: // event accepted
-                case 410: // instance completed or failed
-                    return;
-                case 404:
-                    return Promise.reject(new Error(`No instance with ID '${instanceId}' found.`));
-                case 400:
-                    return Promise.reject(new Error("Only application/json request content is supported"));
-                default:
-                    return Promise.reject(new Error(`Webhook returned unrecognized status code ${response.status}`));
+            if (connectionName) {
+                requestUrl = requestUrl.replace(/(connection=)([\w]+)/gi, "$1" + connectionName);
             }
-        } catch (error) {   // error object is axios-specific, not a JavaScript Error; extract relevant bit
-            throw error.message;
+        }
+
+        const response = await this.axiosInstance.post(requestUrl, JSON.stringify(eventData));
+        switch (response.status) {
+            case 202: // event accepted
+            case 410: // instance completed or failed
+                return;
+            case 404:
+                return Promise.reject(new Error(`No instance with ID '${instanceId}' found.`));
+            case 400:
+                return Promise.reject(new Error("Only application/json request content is supported"));
```

Comment:
I know this isn't new code, but technically a 400 can also indicate invalid JSON content. Why don't we just make the Error message match the body of the response?

Reply:
I'm not confident about how to reliably transform the response content into a string that can be included in the error message, so I don't feel super comfortable making that change as part of this PR. I'm okay with updating the error message, though.

Reply:
Ah, I didn't realize we embed the message as a JSON object. I would say we can just punt that for now.